hubblestack/extmods/returners/sqlite.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+'''
+Return hubble data to sqlite (intended for testing)
+'''
+from __future__ import absolute_import
+
+import salt.returners
+import logging
+import os
+
+__virtualname__ = 'sqlite'
+
+log = logging.getLogger(__virtualname__)
+
+_conn = None
+
+def __virtual__():
+    '''
+    Return virtual name of the module.
+
+    :return: The virtual name of the module.
+    '''
+    global json
+    global sqlite3
+    try:
+        import json
+        GOT_JSON = True
+    except:
+        pass
+    try:
+        import sqlite3
+        GOT_SQLI = True
+    except:
+        pass
+    if GOT_JSON and GOT_SQLI:
+        return __virtualname__
+    msgs = []
+    if not GOT_JSON:
+        msgs.append("json module is missing")
+    if not GOT_SQLI:
+        msgs.append("sqlite3 module is missing")
+    return False, ', '.join(msgs)
```

Comment:
The "salty" pattern for gated imports is to do these try:except blocks outside the `__virtual__`. Can you move these imports accordingly?

Reply:
I could change that and have no problem doing it. But I don't understand why. Moving them to global space means polluting the global() dict with a bunch of ALL_CAPS variables so you can generate a meaningful error. Even if you import more than once by mistake, the import just fails to actually import, so you don't lose anything.

Is the motivation pure distaste for the 'global' keyword or something else?

Reply:
(Though, the more I think about it... because of the way I'm doing it, it might just fool python into re-importing every __virtual__() call ... not really sure.)

Reply:
If we're going to bring code lint into it I still think it belongs outside, so it's grouped with the rest of the imports.

Reply:
That also has the nice side effect of removing the need for the `global` keyword, which is nice.